In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
%cd /content/drive/MyDrive/Causal_Inference_Model/

/content/drive/MyDrive/Causal_Inference_Model


In [ ]:
import warnings
warnings.filterwarnings('ignore')

## Import Libraries

In [ ]:
import dvc.api
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.graph_objects as go
import sys, os

In [45]:
sys.path.append(os.path.abspath('scripts/'))

In [56]:
from filehundle import LoadData
from plots import Plot
from overview import Overview
from preprocessing import PreProcess

Initialize Loaded Data

In [58]:
loader = LoadData()
plot = Plot()

INFO:logger:Successfully Instantiated load_data Class Object
INFO:logger:Successfully Instantiated Preprocessing Class Object


In [59]:
# Get URL from DVC
data_path = 'data/nb.csv'
repo = 'https://github.com/yonamg/Causal_Inference_Model/'
version = '31a6d9ca5315cecd1bf0cfee00a7e0'

In [ ]:
data_url = dvc.api.get_url(
    path=data_path,
    repo=repo,
    rev=version
)

In [62]:
trip_df = loader.read_csv("data/nb.csv")


In [63]:
trip_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 536020 entries, 0 to 536019
Data columns (total 5 columns):
 #   Column            Non-Null Count   Dtype 
---  ------            --------------   ----- 
 0   Trip ID           536020 non-null  int64 
 1   Trip Origin       536020 non-null  object
 2   Trip Destination  536020 non-null  object
 3   Trip Start Time   534369 non-null  object
 4   Trip End Time     536019 non-null  object
dtypes: int64(1), object(4)
memory usage: 20.4+ MB


In [64]:
miss_info = trip_df.isna().sum()
miss_info

Trip ID                0
Trip Origin            0
Trip Destination       0
Trip Start Time     1651
Trip End Time          1
dtype: int64

In [65]:
trip_df.rename(columns = {'Trip ID':'order_id',
                          'Trip Origin':"trip_origin",
                          'Trip Destination':'trip_destination',
                          'Trip Start Time':'trip_Start_time',
                          'Trip End Time':'trip_end_time'}, inplace = True)

In [66]:
trip_df.head()

order_id                         trip_origin  \
0    391996  6.508813001668548,3.37740316890347   
1    391997                 6.4316714,3.4555375   
2    391998         6.631679399999999,3.3388976   
3    391999         6.572757200000001,3.3677082   
4    392001                 6.6010417,3.2766339   

                  trip_destination      trip_Start_time        trip_end_time  
0      6.650969799999999,3.3450307  2021-07-01 07:28:04  2021-07-01 07:29:37  
1  6.4280814653326,3.4721885847586  2021-07-01 06:38:04  2021-07-01 07:07:28  
2      6.508324099999999,3.3590397  2021-07-01 06:21:02  2021-07-01 07:02:23  
3      6.584881099999999,3.3614073  2021-07-01 07:16:07  2021-07-01 07:29:42  
4              6.4501069,3.3916154  2021-07-01 09:30:59  2021-07-01 09:34:36

There are 1651 Trip which has no record of trip start time but has trip end time. Why the is a record for trip end time but not for trip start time? Do the driver register their end time only to do some tricks or is there system error for those sptecific trip without trip start time?